### PR TITLE
Update certificate bundle; update docker builds

### PR DESCRIPTION
**Pull request by fcpranav**
## Description
We learned that a recent cert upgrade caused problems for some people who use pdagent. The changes in this pr accomplish the following

- [x] Certs are updated
- [x] Both the `ubuntu` and `centos` builds now run
- [x] If people want to run builds inside Docker (which does not support `systemd` although it technically exists), there is now a way to specify the correct option so that the packages can be Docker-compatible
- [x] Documentation is updated 
- [x] Update version number
- [x] Update tests so that they run in python3 (remove throttling testing)

### CODE REVIEW

File: build-linux/howto.md [hunk 1]
```diff
@@ -10,16 +10,20 @@ To build Linux packages, you will need GPG v1 keys to sign the packages.  Do the
 
 ```
 brew install gpg1
-mkdir build-linux/gnupg
-chmod 700 build-linux/gnupg
-gpg1 --homedir=build-linux/gnupg --gen-key
+mkdir build-linux/gpg-deb
```
Comment (hcopperm) [inline build-linux/howto.md:13]: reminder for us: this change will work for the deb package but not the rpm one i think

Comment (hcopperm) [reply build-linux/howto.md:13]: working that out now

File: Dockerfile-centos-7 [hunk 2]
```diff
@@ -3,10 +3,14 @@ FROM centos/systemd
 ARG FPM_VERSION=1.11.0
 ARG PYTHON_VERSION=3
 ARG DOCKER_WORKDIR=/usr/share/pdagent
+ARG SKIP_SYSTEMD=false
 ENV PYTHON_VERSION ${PYTHON_VERSION}
 ENV container docker
 ENV DOCKER_WORKDIR ${DOCKER_WORKDIR}
 
+# There is known issues with Docker + systemd
```
Comment (joshwertheim) [inline Dockerfile-centos-7:11]: Did we only want to include this for testing? Is it helpful for it to remain in the changes or at least not commented out for people who will inevitably build from source?

Comment (joshwertheim) [reply Dockerfile-centos-7:11]: ahhh disregard I hadn't seen you have a config boolean for this.

**Review by joshwertheim — APPROVED** [review]
tentative approval, would ask for one more approval before merging.

File: Dockerfile-ubuntu [hunk 3]
```diff
@@ -25,6 +30,13 @@ RUN rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
 RUN update-ca-certificates
 
+
+# Lock dotenv version to prevent ruby 3.0 requirement
```
Comment (acloudgopher-pd) [inline Dockerfile-ubuntu:34]: I'd apply this same comment to the `centos` one if it applies

Comment (hcopperm) [reply Dockerfile-ubuntu:34]: oh yeah it does, good call

File: Dockerfile-centos-7 (hunk 2 above)
Comment (blagh) [inline Dockerfile-centos-7:11]:
```suggestion
# There are known compatibility issues with Docker + systemd -- they both assume they are running with pid 1
```

plus maybe a handy link if we have it?

File: build-linux/pdagent.init [hunk 4]
```diff
@@ -60,7 +60,7 @@ get_pid() {
 
 is_running() {
   pid=$(get_pid)
-  [ -n "$pid" ] && ps -p $pid | grep "$EXEC_NAME" >/dev/null 2>&1
+  [ -n "$pid" ] && ps $pid | grep "$EXEC_NAME" >/dev/null 2>&1
```
Comment (acloudgopher-pd) [inline build-linux/pdagent.init:63]: Curious why -p is dropped?

**Review by blagh — APPROVED** [review]
nits only

File: build-linux/pdagent.init (hunk 4 above)
Comment (hcopperm) [reply build-linux/pdagent.init:63]: for some reason (we couldn't find documentation about why) `-p` wasn't actually displaying the full name of the process that was being run, so this no longer worked. for example:
`ps $pid` => `python /path/to/pdagent.py` => status: running
`ps -p $pid` => `python`=> status: not running

**Review by acloudgopher-pd — APPROVED** [review]
Looks ok to me, I might suggest changing the PR title a bit to make it more descriptive

File: unit_tests/test_pdqueue.py [hunk 5]
```diff
@@ -792,6 +792,7 @@ def stats_by_state(data):
                 "started_on": "some_utc_time"
                 }
             }
+        self.maxDiff = None
```
[the remark this answers is not in the document]
Comment (hcopperm) [reply unit_tests/test_pdqueue.py:795]: i'll add a note

**Comment by hcopperm** [comment]
> Looks ok to me, I might suggest changing the PR title a bit since it is a public repo

good idea; i removed internal links too

File: build-linux/pdagent.init (hunk 4 above)
Comment (acloudgopher-pd) [reply build-linux/pdagent.init:63]: got it, thank you!

File: Dockerfile-centos-7 (hunk 2 above)
Comment (hcopperm) [reply Dockerfile-centos-7:11]: i can't judge whether we do need this much detail.......

Comment (blagh) [reply Dockerfile-centos-7:11]: Eh: I feel like it's helpful to include the concrete reason why, it's not a trade secret or anything.

**Comment by wlonkly** [comment]
The update-certs branch was included in customer comms during the pdagent/Digicert rotation situation, please leave this branch in place!

**Pull request merged**